azurerm/internal/services/loganalytics/log_analytics_datasource_linux_syslog_resource.go

```diff
@@ -0,0 +1,226 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/set"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/loganalytics/parse"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/state"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmLogAnalyticsDataSourceLinuxSyslog() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLogAnalyticsDataSourceLinuxSyslogCreateUpdate,
+		Read:   resourceArmLogAnalyticsDataSourceLinuxSyslogRead,
+		Update: resourceArmLogAnalyticsDataSourceLinuxSyslogCreateUpdate,
+		Delete: resourceArmLogAnalyticsDataSourceLinuxSyslogDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImportThen(func(id string) error {
+			_, err := parse.LogAnalyticsDataSourceID(id)
+			return err
+		}, importLogAnalyticsDataSource(operationalinsights.LinuxSyslog)),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"workspace_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     ValidateAzureRmLogAnalyticsWorkspaceName,
+			},
+
+			"syslog_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"auth",
+					"authpriv",
+					"cron",
+					"daemon",
+					"ftp",
+					"kern",
+					"mail",
+					"syslog",
+					"user",
+					"uucp",
+				}, false),
+			},
+
+			"syslog_severities": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MinItems: 1,
+				Set:      set.HashStringIgnoreCase,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					// API backend accepts event_types case-insensitively
+					ValidateFunc:     validation.StringInSlice([]string{"emerg", "alert", "crit", "err", "warning", "notice", "info", "debug"}, true),
+					StateFunc:        state.IgnoreCase,
+					DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
You should be able to remove this if you are pulling the values from the SDK?

Reply:
Due to no enum support from az-sdk-for-go, can not remove them